WordPress/Classes/Utility/Media/MediaAssetExporter.swift

```diff
@@ -50,172 +50,198 @@ class MediaAssetExporter: MediaExporter {
+                                    onCompletion(imageExport)
+                                },
+                                                onError: onError)
+        })
+        return progress
     }
 
     /// Exports and writes an asset's video data to a local Media URL.
     ///
     /// - parameter onCompletion: Called on successful export, with the local file URL of the exported asset.
     /// - parameter onError: Called if an error was encountered during export.
     ///
-    fileprivate func exportVideo(forAsset asset: PHAsset, onCompletion: @escaping OnMediaExport, onError: @escaping OnExportError) {
-        do {
-            guard asset.mediaType == .video else {
-                throw AssetExportError.expectedPHAssetVideoType
-            }
-            // Get the resource matching the type, to export.
-            let resources = PHAssetResource.assetResources(for: asset).filter({ $0.type == .video })
-            guard let videoResource = resources.first else {
-                throw AssetExportError.unavailablePHAssetVideoResource
-            }
+    fileprivate func exportVideo(forAsset asset: PHAsset, onCompletion: @escaping OnMediaExport, onError: @escaping OnExportError) -> Progress {
+        guard asset.mediaType == .video else {
+            onError(exporterErrorWith(error: AssetExportError.expectedPHAssetVideoType))
+            return Progress.discreteCompletedProgress()
+        }
+        // Get the resource matching the type, to export.
+        let resources = PHAssetResource.assetResources(for: asset).filter({ $0.type == .video })
+        guard let videoResource = resources.first else {
+            onError(exporterErrorWith(error: AssetExportError.unavailablePHAssetVideoResource))
+            return Progress.discreteCompletedProgress()
+        }
 
-            // Configure a video exporter to handle an export session.
-            let videoExporter = MediaVideoExporter()
-            videoExporter.mediaDirectoryType = mediaDirectoryType
+        // Configure a video exporter to handle an export session.
+        var exporterVideoOptions = videoOptions ?? MediaVideoExporter.Options()
 
-            if let options = videoOptions {
-                videoExporter.options = options
-            }
-            if videoExporter.options.preferredExportVideoType == nil {
-                videoExporter.options.preferredExportVideoType = videoResource.uniformTypeIdentifier
+        if exporterVideoOptions.preferredExportVideoType == nil {
+            exporterVideoOptions.preferredExportVideoType = videoResource.uniformTypeIdentifier
+        }
+
+        let originalFilename = videoResource.originalFilename
+
+        // Request an export session, which may take time to download the complete video data.
+        let options = PHVideoRequestOptions()
+        options.isNetworkAccessAllowed = true
+        let progress = Progress.discreteProgress(totalUnitCount: 100)
```

Comment:
See earlier comments for these 100 / 50 values. Perhaps we can even have some helper methods on `Progress` to encapsulate what this is doing.